### ISSUE TITLE
renderToString incorrectly supports select's defaultValue

### ISSUE THREAD

**Issue by nkt** [facebook/react#4333]
[Here is code for reproducing:](https://jsfiddle.net/69z2wepo/11806/)
```js
var Test = React.createClass({
  render: function() {
    return (
      <select defaultValue={this.props.current}>
        {this.props.items.map((i) => (
          <option value={i} key={i}>{i}</option>
        ))}
      </select>
    );
  }
});

var container = document.getElementById('container');
var code = document.getElementById('code');
var test = <Test items={[1, 2, 3]} current={2} />;
var prerender = React.renderToString(test);
code.innerText = prerender.replace(/></g, '>\n<');

container.innerHTML = prerender;
setTimeout(() => {
  React.render(test, container)
}, 1000);
```
So I'm rendering `select` on  the server and got small lag on the client.

**Comment by jimfb** [comment]
Well, that's annoying.  https://jsfiddle.net/pgtf6aym/

**Comment by zpao** [comment]
Fixed in 0.14 by #3595.
